specs/eventing/data-plane.md

```diff
@@ -1,125 +1,179 @@
-sent again. If the event can not be delivered then some sources of events (such
-as Knative sources, brokers or channels) MAY support
-[dead letter sink or channel](https://github.com/knative/eventing/blob/main/docs/delivery/README.md) for events that can not be
-delivered.
+All senders and recipients MUST support the CloudEvents 1.0 protocol and the
+[binary](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#31-binary-content-mode)
+and
+[structured](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#32-structured-content-mode)
+content modes of the CloudEvents HTTP binding. Senders which do not advertise
+the ability to accept [reply events](#derived-reply-events) MAY implement only
+one content mode, as the recipient is not allowed to negotiate the content mode.
+Senders MUST support both cleartext (`http`) and TLS (`https`) URLs as event
+delivery destinations.
+
+### HTTP Verbs
+
+In the absence of specific delivery preferences, the sender MUST initiate
+delivery of the event to the recipient using the HTTP POST verb, using either
+the structured or binary encoding of the event (sender's choice). This delivery
+SHOULD be performed using the CloudEvents HTTP Binding, version 1.0.
+
+Senders MAY probe the recipient with an
+[HTTP OPTIONS request](https://tools.ietf.org/html/rfc7231#section-4.3.7); if
+implemented, the recipient MUST indicate support for the POST verb using the
+[`Allow` header](https://tools.ietf.org/html/rfc7231#section-7.4.1). Senders
+which receive an error when probing with HTTP OPTIONS SHOULD proceed using the
+HTTP POST mechanism.
+
+### Event Acknowledgement and Delivery Retry
+
+Event recipients MUST use the HTTP response code to indicate acceptance of an
+event. The recipient SHOULD NOT return a response accepting the event until it
+has handled event (processed the event or stored it in stable storage). The
+following response codes are explicitly defined; event recipients MAY also
+respond with other response codes. A response code not in this table SHOULD be
+treated as a retriable error.
+
+| Response code | Meaning                           | Retry | Delivery completed | Error |
+| ------------- | --------------------------------- | ----- | ------------------ | ----- |
+| `1xx`         | (Unspecified)                     | No\*  | No\*               | Yes\* |
+| `200`         | [Event reply](#event-reply)       | No    | Yes                | No    |
+| `202`         | Event accepted                    | No    | Yes                | No    |
+| other `2xx`   | (Unspecified)                     | No\*  | No\*               | Yes\* |
+| other `3xx`   | (Unspecified)                     | No\*  | No\*               | Yes\* |
+| `400`         | Unparsable event                  | No    | No                 | Yes   |
+| `404`         | Endpoint does not exist           | Yes   | No                 | Yes   |
```

Comment:
Does a Yes in the Retry column means that the sender MUST retry or that it MAY retry if it wants. I'm not sure I would want to mandate a Retry on a 404.

Reply:
Or on any error TBH

Reply:
L134 specifies this as a "SHOULD":

> Where possible, event senders SHOULD re-attempt delivery of events where the HTTP request failed or returned a retryable status code.

Reply:
I'm still uncomfortable with a SHOULD for a 404 and 5xx errors. I can buy an argument that 429 might be retried, but the others (and probably 409) don't give me any reason to believe that a subsequent replay of the event would result in any different result.  Maybe 404 if you believe the sink is still coming up I guess, but all of the others are just asking to be DDOSs

Reply:
What about a MAY if the sender has reason to believe that a retry might yield different results?